lambdas/checkpassport/src/main/java/uk/gov/di/ipv/cri/passport/checkpassport/services/DocumentDataVerificationService.java

```diff
@@ -0,0 +1,164 @@
+import uk.gov.di.ipv.cri.common.library.persistence.item.SessionItem;
+import uk.gov.di.ipv.cri.common.library.service.AuditService;
+import uk.gov.di.ipv.cri.common.library.util.EventProbe;
+import uk.gov.di.ipv.cri.passport.checkpassport.domain.result.DocumentDataVerificationResult;
+import uk.gov.di.ipv.cri.passport.checkpassport.domain.result.ThirdPartyAPIResult;
+import uk.gov.di.ipv.cri.passport.checkpassport.exception.OAuthHttpResponseExceptionWithErrorBody;
+import uk.gov.di.ipv.cri.passport.checkpassport.validation.ValidationResult;
+import uk.gov.di.ipv.cri.passport.library.domain.PassportFormData;
+import uk.gov.di.ipv.cri.passport.library.error.ErrorResponse;
+import uk.gov.di.ipv.cri.passport.library.helpers.PersonIdentityDetailedHelperMapper;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.di.ipv.cri.passport.library.domain.CheckType.DOCUMENT_DATA_VERIFICATION;
+import static uk.gov.di.ipv.cri.passport.library.metrics.Definitions.DOCUMENT_DATA_VERIFICATION_REQUEST_FAILED;
+import static uk.gov.di.ipv.cri.passport.library.metrics.Definitions.DOCUMENT_DATA_VERIFICATION_REQUEST_SUCCEEDED;
+import static uk.gov.di.ipv.cri.passport.library.metrics.Definitions.FORM_DATA_VALIDATION_FAIL;
+import static uk.gov.di.ipv.cri.passport.library.metrics.Definitions.FORM_DATA_VALIDATION_PASS;
+
+public class DocumentDataVerificationService {
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    private static final int MAX_PASSPORT_GPG45_STRENGTH_VALUE = 4;
+    private static final int MAX_PASSPORT_GPG45_VALIDITY_VALUE = 2;
+    private static final int MIN_PASSPORT_GPG45_VALUE = 0;
+
+    private final EventProbe eventProbe;
+    private final AuditService auditService;
+
+    private final FormDataValidator formDataValidator;
+    private final ThirdPartyAPIService thirdPartyAPIService;
+
+    public DocumentDataVerificationService(
+            EventProbe eventProbe,
+            AuditService auditService,
+            ThirdPartyAPIService thirdPartyAPIService,
+            FormDataValidator formDataValidator) {
+
+        this.eventProbe = eventProbe;
+        this.auditService = auditService;
+
+        this.thirdPartyAPIService = thirdPartyAPIService;
+        this.formDataValidator = formDataValidator;
+    }
+
+    public DocumentDataVerificationResult verifyData(
+            PassportFormData passportFormData,
+            SessionItem sessionItem,
+            Map<String, String> requestHeaders)
+            throws OAuthHttpResponseExceptionWithErrorBody {
+        try {
+            LOGGER.info("Validating form data...");
+            ValidationResult<List<String>> validationResult =
+                    this.formDataValidator.validate(passportFormData);
+            if (!validationResult.isValid()) {
+                String errorMessages = String.join(",", validationResult.getError());
+                LOGGER.error(
```

Comment:
Theres also a potential we log PII here but its shouldnt really be possible to get here with bad data but its always better to be safe